pkg/epp/datastore/datastore.go

```diff
@@ -89,6 +95,8 @@ type datastore struct {
 	pool                *v1.InferencePool
 	// key: InferenceObjective name, value: *InferenceObjective
 	objectives map[string]*v1alpha2.InferenceObjective
+	// key: types.NamespacedName, value: *v1alpha2.InferenceModelRewrite
+	rewrites map[types.NamespacedName]*v1alpha2.InferenceModelRewrite
```

Comment:
Ideally I don't want to fetch ALL Rewrite objects on every request (assuming there will be objects that are model specific).
I think this should be reflected here in the way we store the objects. 
keeping NamespacedName as the key in the name doesn't give us any "smart" mapping.
I think we should map from model_name -> rewrite object, while keeping empty model name as match for all requests. so on every request we call datastore to get the relevant rewrite objects and get back the:
- specific rewrite objects for the requested model; and
- the general rewrite objects that match all requests.